--- a/src/dvb/dvbdevice.cpp
+++ b/src/dvb/dvbdevice.cpp
@@ -353,8 +353,8 @@
 
 	switch (config->configuration) {
 	case DvbConfigBase::DiseqcSwitch: {
-		char cmd[] = { 0xe0, 0x10, 0x38, 0x00 };
-		cmd[3] = 0xf0 | (config->lnbNumber << 2) | (horPolar ? 2 : 0) | (highBand ? 1 : 0);
+		char cmd[] = { char(0xe0), 0x10, 0x38, 0x00 };
+		cmd[3] = 0xf0 | char(config->lnbNumber << 2) | (horPolar ? 2 : 0) | (highBand ? 1 : 0);
 		backend->sendMessage(cmd, sizeof(cmd));
 		usleep(15000);
 
@@ -399,7 +399,7 @@
 			value |= 0xd000;
 		}
 
-		char cmd[] = { 0xe0, 0x31, 0x6e, (value / 256), (value % 256) };
+		char cmd[] = { char(0xe0), 0x31, 0x6e, char(value / 256), char(value % 256) };
 		backend->sendMessage(cmd, sizeof(cmd));
 		usleep(15000);
 		moveRotor = true;
@@ -407,7 +407,7 @@
 	    }
 
 	case DvbConfigBase::PositionsRotor: {
-		char cmd[] = { 0xe0, 0x31, 0x6b, config->lnbNumber };
+		char cmd[] = { char(0xe0), 0x31, 0x6b, char(config->lnbNumber) };
 		backend->sendMessage(cmd, sizeof(cmd));
 		usleep(15000);
 		moveRotor = true;

